packages/opentelemetry-plugin-http/src/utils.ts

```diff
@@ -63,32 +63,28 @@ export const getAbsoluteUrl = (
 export const parseResponseStatus = (
   statusCode: number
 ): Omit<Status, 'message'> => {
-  if (statusCode < 200 || statusCode > 504) {
-    return { code: CanonicalCode.UNKNOWN };
-  } else if (statusCode >= 200 && statusCode < 400) {
+  if (statusCode < 400) {
     return { code: CanonicalCode.OK };
-  } else {
-    switch (statusCode) {
-      case 400:
-        return { code: CanonicalCode.INVALID_ARGUMENT };
-      case 504:
-        return { code: CanonicalCode.DEADLINE_EXCEEDED };
-      case 404:
-        return { code: CanonicalCode.NOT_FOUND };
-      case 403:
-        return { code: CanonicalCode.PERMISSION_DENIED };
-      case 401:
-        return { code: CanonicalCode.UNAUTHENTICATED };
-      case 429:
-        return { code: CanonicalCode.RESOURCE_EXHAUSTED };
-      case 501:
-        return { code: CanonicalCode.UNIMPLEMENTED };
-      case 503:
-        return { code: CanonicalCode.UNAVAILABLE };
-      default:
-        return { code: CanonicalCode.UNKNOWN };
-    }
+  } else if (statusCode === 404) {
```

Comment:
I think **switch statemen**t would look more efficient here to eliminate redundant code. WDYT?

Reply:
from my gist link here the example:

```
const parseResponseStatus = (
  statusCode
) => {
  if (statusCode < 400) {
    return { code: CanonicalCode.OK };
  } else {
    switch (statusCode) {
      case 404:
        return { code: CanonicalCode.NOT_FOUND };
      case 403:
        return { code: CanonicalCode.PERMISSION_DENIED };
      case 401:
        return { code: CanonicalCode.UNAUTHENTICATED };
      case 429:
        return { code: CanonicalCode.RESOURCE_EXHAUSTED };
      case 501:
        return { code: CanonicalCode.UNIMPLEMENTED };
      case 503:
        return { code: CanonicalCode.UNAVAILABLE };
      case 504:
        return { code: CanonicalCode.DEADLINE_EXCEEDED };
      default:
        if (statusCode < 500) {
          return { code: CanonicalCode.INVALID_ARGUMENT };
        } else if (statusCode < 512 || statusCode === 598 || statusCode === 599) {
          return { code: CanonicalCode.INTERNAL };
        }
        return { code: CanonicalCode.UNKNOWN };
    }
  }
};
```

Do you prefer this ? Benchmark tool wasn't really helpful to determine which one is faster (see gist link for the file)...
Do you have some tips that I could try ? Thanks

Reply:
Because object keys can be numbers, what would you think of this:

```typescript
const httpStatusSpecialCases = {
	401: CanonicalCode.UNAUTHENTICATED,
	403: CanonicalCode.PERMISSION_DENIED,
	404: CanonicalCode.NOT_FOUND,
	429: CanonicalCode.RESOURCE_EXHAUSTED,

	598: CanonicalCode.INTERNAL,
	599: CanonicalCode.INTERNAL,
}

function parseResponseStatus(statusCode: number) {
	// Handled as a special case?
	const code = httpStatusSpecialCases[statusCode];
    if (code !== undefined) { return { code }; }

	// In OK range?
	if (code >= 100 && code < 400) {
		return { code: CanonicalCode.OK };
	}

	// Client errors
	if (code >= 400 && code < 500) {
		return { code: CanonicalCode.INVALID_ARGUMENT };
	}

	// Internal errors
	if (code >= 500 && code < 512) {
		return { code: CanonicalCode.INTERNAL };
	}

	// All other codes are unknown
	return { code: CanonicalCode.UNKNOWN };
}
```

Reply:
It could even be further simplified:

```typescript
const httpStatusSpecialCases = {
	401: CanonicalCode.UNAUTHENTICATED,
	403: CanonicalCode.PERMISSION_DENIED,
	404: CanonicalCode.NOT_FOUND,
	429: CanonicalCode.RESOURCE_EXHAUSTED,

	// etc.

	598: CanonicalCode.INTERNAL,
	599: CanonicalCode.INTERNAL,
}

function parseResponseStatus(statusCode: number) {
	// Handled as a special case?
	const code = httpStatusSpecialCases[statusCode];
    if (code !== undefined) { return { code }; }

	// 0xx are unknown
	if (code < 100) {
		return { code: CanonicalCode.UNKNOWN };
	}

	// 1xx, 2xx, 3xx are OK
	if (code < 400) {
		return { code: CanonicalCode.OK };
	}

	// 4xx are client errors
	if (code < 500) {
		return { code: CanonicalCode.INVALID_ARGUMENT };
	}

	// 5xx are internal errors
	if (code < 512) {
		return { code: CanonicalCode.INTERNAL };
	}

	// All other codes are unknown
	return { code: CanonicalCode.UNKNOWN };
}
```